### PR TITLE
Outlook msg fix

### PR DESCRIPTION
**Pull request by winiar93**
## Summary
Workaround that solves outlook messages extraction issue.

## Checklist

This PR:

- [X] adds new tests (if appropriate)
- [X] updates docstrings for any new functions or function arguments (if appropriate)
- [X] updates `CHANGELOG.md` with a summary of the changes

### CODE REVIEW

File: viadot/sources/outlook.py [hunk 1]
```diff
@@ -82,7 +88,8 @@ def __init__(
             print(f"{self.mailbox_name} NOT Authenticated!")
 
         self.mailbox_obj = self.account.mailbox()
-        self.mailbox_messages = self.mailbox_obj.get_messages(limit)
+        self.mailbox_folders = mailbox_folders
```
Comment (angelika233) [inline viadot/sources/outlook.py:91]: You can add check if mailbox_folders is in allowed values

File: viadot/sources/outlook.py [hunk 2]
```diff
@@ -92,66 +99,71 @@ def to_df(self) -> pd.DataFrame:
             pd.DataFrame: the DataFrame with time range
```
Comment (angelika233) [inline viadot/sources/outlook.py:99]: Add dot at the end of the sentence

File: viadot/sources/outlook.py [hunk 3]
```diff
@@ -92,66 +99,71 @@ def to_df(self) -> pd.DataFrame:
...
+            base_str = f"self.mailbox_obj.{m}_folder().get_messages({self.limit})"
+            mailbox_generators_list.append(eval(base_str))
+
+        for mailbox_generator in mailbox_generators_list:
+            while True:
+                try:
+                    message = next(mailbox_generator)
+                    received_time = message.received
+                    date_time_str = str(received_time)
+                    date_time_string = date_time_str[0:19]
```
Comment (angelika233) [inline viadot/sources/outlook.py:113]: Maybe change the name of variable. It's okay, but it's almost the same as `date_time_str`. I would suggest sth like `date_time_stripped` or `date_time_trimmed`.

File: viadot/sources/outlook.py [hunk 4]
```diff
@@ -92,66 +99,71 @@ def to_df(self) -> pd.DataFrame:
...
+                    if (
+                        date_obj < self.date_range_start_time
+                        or date_obj > self.date_range_end_time
+                    ):
+                        continue
+                    else:
+                        fetched = message.to_api_data()
+                        try:
+                            sender_mail = fetched["from"]["emailAddress"]["address"]
+                            reciver_list = fetched.get("toRecipients")
```
Comment (angelika233) [inline viadot/sources/outlook.py:126]: Small suggestion: maybe `recivers_list` would be better?

**Review by angelika233 — CHANGES_REQUESTED** [review]
small suggestions

File: viadot/sources/outlook.py [hunk 5]
```diff
@@ -34,6 +37,9 @@ def __init__(
             ACCOUNT_NAME and Service Principal credentials (TENANT_ID, CLIENT_ID, CLIENT_SECRET) for the Azure Application.
             Defaults to None.
             limit (int, optional): Number of fetched top messages. Defaults to 10000.
+            mailbox_folders (Literal["sent", "inbox", "junk", "deleted", "drafts", "outbox", "archive"]):
+                List of folders to select from the mailbox.  Defaults to ["sent", "inbox", "junk", "deleted", "drafts", "outbox", "archive"]
+            request_retries (int): How many times retries to authorizate. Defaults to 10.
```
Comment (Rafalz13) [inline viadot/sources/outlook.py:42]: Add optional - (int, `optional`)

File: viadot/sources/outlook.py [hunk 6]
```diff
@@ -82,76 +88,96 @@ def __init__(
...
-            except StopIteration:
-                break
+                            if sender_mail == self.mailbox_name:
+                                row["Inbox"] = False
+                            else:
+                                row["Inbox"] = True
+
+                            data.append(row)
+                        except KeyError as e:
+                            print("KeyError : " + str(e))
```
Comment (Rafalz13) [inline viadot/sources/outlook.py:178]: replace `print` with `logger`

File: viadot/sources/outlook.py [hunk 7]
```diff
@@ -34,6 +37,9 @@ def __init__(
             ACCOUNT_NAME and Service Principal credentials (TENANT_ID, CLIENT_ID, CLIENT_SECRET) for the Azure Application.
             Defaults to None.
             limit (int, optional): Number of fetched top messages. Defaults to 10000.
+            mailbox_folders (Literal["sent", "inbox", "junk", "deleted", "drafts", "outbox", "archive"]):
+                List of folders to select from the mailbox.  Defaults to ["sent", "inbox", "junk", "deleted", "drafts", "outbox", "archive"]
+            request_retries (int): How many times retries to authorizate. Defaults to 10.
         """
         try:
             DEFAULT_CREDENTIALS = local_config["OUTLOOK"]
```
Comment (Rafalz13) [inline viadot/sources/outlook.py:45]: you can change the authentication method. Use `credentials_loader`.

**Review by Rafalz13 — CHANGES_REQUESTED** [review]

**Review by Rafalz13 — APPROVED** [review]

**Pull request merged**